esphome/components/esp32/__init__.py

```diff
@@ -172,6 +172,24 @@ def add_idf_component(
             KEY_COMPONENTS: components,
             KEY_SUBMODULES: submodules,
         }
+    else:
+        if components is not None:
```

Comment:
Lets pull this out to clean it up a bit
```suggestion
        component_config = CORE.data[KEY_ESP32][KEY_COMPONENTS][name]
        if components is not None:
```